Helpers/get_results_from_criteria_and_data.py

```diff
@@ -0,0 +1,78 @@
+from datetime import date, timedelta
+from typing import List
+
+from graphql_objects import DatesCheck
+
+
+class GetResultsFromSearchCriteriaAndData():
```

Comment:
Moving everything into this class did not solve the OCP problem. The OCP problem asks, when you do the for loop through your list of houses, is this extensible to different kinds of houses you have in the future? What happens if you have an `ApartmentComplex` or wtv that needs a different available algo to get the resultant `DatesCheck`?

The answer is that much of this code actually belongs in the `Houses` object.